src/gui/FileBrowser.cpp

```diff
@@ -328,68 +328,62 @@ void FileBrowser::addItems(const QString & path )
 	}
 
 	// try to add all directories from file system alphabetically into the tree
-	QDir cdir( path );
-	QStringList files = cdir.entryList( QDir::Dirs, QDir::Name );
-	files.sort(Qt::CaseInsensitive);
-	for( QStringList::const_iterator it = files.constBegin();
-						it != files.constEnd(); ++it )
+	QDir cdir(path);
+	if (!cdir.isReadable()) { return; }
+	cdir.setNameFilters(m_filter.split(' '));
+	QFileInfoList entries = cdir.entryInfoList(
+			QDir::AllDirs | QDir::Files | QDir::NoDotAndDotDot,
+			QDir::LocaleAware | QDir::DirsFirst | QDir::Name | QDir::IgnoreCase);
```

Comment:
I believe `entryInfoList` has an overload taking a list of name filters, so they don't need to be set separately.